script.js

```diff
@@ -1,12 +1,123 @@
+        episodeName.style.textAlign = "center";
+        episodeName.style.marginTop = "0px";
+        episodeElm.appendChild(episodeName);
+        // The Image
+        var episodeImg = document.createElement("img");
+        episodeImg.src = episodeList[i].image.medium;
+        episodeImg.style.alignSelf = "center";
+        episodeImg.style.margin = "auto";
+        episodeImg.style.display = "block";
+        //episodeImg.style.width = "50%";
+        episodeElm.appendChild(episodeImg);
+        //The Episode Summary
+        var episodesummary = document.createElement("div");
+        episodesummary.innerHTML = episodeList[i].summary;
+        episodesummary.style.padding = "10px";
+        episodeElm.appendChild(episodesummary);
+
+        //rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+    }
+}
+
+function liveSearch() {
+    newEpisodes = [];
+
+    var searchResult = document.getElementById("searchResult");
+
+    const searchString = searchInput.value.toLowerCase();
+    const filteredEpisodes = allEpisodes.filter((episode) => {
+        return (
+            episode.summary.toLowerCase().includes(searchString) ||
+            episode.name.toLowerCase().includes(searchString)
+        );
+    });
+    return (searchResult.innerText =
+        "Displaying " + filteredEpisodes.length + "/73 episodes");
+
+    // another way to solve the live search
+
+    // for (let i = 0; i < allEpisodes.length; i++) {
+    //     if (
+    //         allEpisodes[i].summary
+    //         .toLowerCase()
+    //         .includes(searchInput.value.toLowerCase()) ||
+    //         allEpisodes[i].name
+    //         .toLowerCase()
+    //         .includes(searchInput.value.toLowerCase())
+    //     )
+    //         newEpisodes.push(allEpisodes[i]);
+    // }
+
+    // return (searchResult.innerText =
+    //   "Displaying " + newEpisodes.length + "/73 episodes");
 }
 
-window.onload = setup;
+function makeInputForEpisodes(episodeList) {
+    // var dataList = document.getElementById("browsers");
+    //pickEpisode.appendChild(dataList);
+    episodeList.map((episode) => {
+        const episodeInput = document.createElement("option");
+        episodeInput.innerText =
+            formatEpisode(episode.season, episode.number) + "-" + episode.name;
+        pickEpisode.appendChild(episodeInput);
+    });
+}
+
+function pickAnEpisod() {
+    const theEpisode = [];
+    rootElem.innerHTML = "";
+    const thePickedEpisod = pickEpisode.selectedIndex;
+    if (thePickedEpisod > 0) {
+        theEpisode.push(allEpisodes[thePickedEpisod - 1]);
+        makePageForEpisodes(theEpisode);
+        const div = document.createElement("div");
+        rootElem.appendChild(div);
+        const returnBut = document.createElement("button");
+        returnBut.innerText = "Return Back";
+        div.appendChild(returnBut);
+        returnBut.addEventListener("click", setup);
+    } else makePageForEpisodes(allEpisodes);
+}
```

Comment:
you have accommodate lots of things in if statement in  pickAnEpisod function. i should have try something like this!!